src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -117,128 +111,23 @@ internal LanguageWorkerChannel()
-            try
-            {
-                if (_process.ExitCode != 0)
-                {
-                    var processExitEx = new LanguageWorkerProcessExitException($"{_process.StartInfo.FileName} exited with code {_process.ExitCode}\n {exceptionMessage}");
-                    processExitEx.ExitCode = _process.ExitCode;
-                    HandleWorkerError(processExitEx);
-                }
-                else
-                {
-                    _process.WaitForExit();
-                    _process.Close();
-                }
-            }
-            catch (Exception)
+            get
             {
-                // ignore process is already disposed
+                return _workerChannelLogger;
             }
-        }
 
-        private void OnErrorDataReceived(object sender, DataReceivedEventArgs e)
-        {
-            // TODO: per language stdout/err parser?
-            if (e.Data != null)
+            set
             {
-                string msg = e.Data;
-                if (msg.IndexOf("warn", StringComparison.OrdinalIgnoreCase) > -1)
-                {
-                    if (LanguageWorkerChannelUtilities.IsLanguageWorkerConsoleLog(msg))
-                    {
-                        msg = LanguageWorkerChannelUtilities.RemoveLogPrefix(msg);
-                        _workerChannelLogger?.LogWarning(msg);
-                    }
-                    else
-                    {
-                        _consoleLogSource?.Log(msg);
-                    }
-                }
-                else if ((msg.IndexOf("error", StringComparison.OrdinalIgnoreCase) > -1) ||
-                          (msg.IndexOf("fail", StringComparison.OrdinalIgnoreCase) > -1) ||
-                          (msg.IndexOf("severe", StringComparison.OrdinalIgnoreCase) > -1))
-                {
-                    if (LanguageWorkerChannelUtilities.IsLanguageWorkerConsoleLog(msg))
-                    {
-                        msg = LanguageWorkerChannelUtilities.RemoveLogPrefix(msg);
-                        _workerChannelLogger?.LogError(msg);
-                    }
-                    else
-                    {
-                        _consoleLogSource?.Log(msg);
-                    }
-                    _processStdErrDataQueue = LanguageWorkerChannelUtilities.AddStdErrMessage(_processStdErrDataQueue, Sanitizer.Sanitize(msg));
-                }
-                else
-                {
-                    if (LanguageWorkerChannelUtilities.IsLanguageWorkerConsoleLog(msg))
-                    {
-                        msg = LanguageWorkerChannelUtilities.RemoveLogPrefix(msg);
-                        _workerChannelLogger?.LogInformation(msg);
-                    }
-                    else
-                    {
-                        _consoleLogSource?.Log(msg);
-                    }
-                }
+                // used for unit tests to incject testLogger
```

Comment:
nit: spelling
Is there a reason why test couldn't pass a factory that would create the appropriate logger? Is this surface really needed?

Reply:
Fixed this in follow up PR